src/Microsoft.VisualStudio.Editors/PropPages/ReferencePropPage.vb

```diff
@@ -792,10 +793,11 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
                 Dim solution = visualStudioWorkspace.CurrentSolution
                 Dim CurrentImportNames As New List(Of String)
 
-                For Each projectId In solution.ProjectIds
-                    ' We need to find the project that matches by IVsHierarchy
-                    If visualStudioWorkspace.GetHierarchy(projectId) Is ProjectHierarchy Then
-                        Dim compilationTask = solution.GetProject(projectId).GetCompilationAsync(cancellationTokenSource.Token)
+                For Each project In solution.Projects
```

Comment:
This is expected to work at the Unconfigured Project level and we do not expect the users to add imports based on the TF that is currently active. There is nothing that prevents the user from doing that but we can discuss how often something like this would happen and handle it then.

To answer your question, for multi target frameworks with different imports for different targetframework(importing conditioned on targetframework), we would pick the project that gets listed first irrespective of its targetframework.